src/kernel/emulation/linux/xattr/flistxattr.c

```diff
@@ -31,7 +31,12 @@ long sys_flistxattr(int fd, char* namebuf, unsigned long size, int options)
 		char buf[64] = {0};
 		char path[4096] = {0};
 		__simple_sprintf(buf, "/proc/self/fd/%d", fd);
-		ret = LINUX_SYSCALL(__NR_readlink, buf, path, sizeof(path) - 1);
+		#if __NR_readlink
+			ret = LINUX_SYSCALL(__NR_readlink, buf, path, sizeof(path) - 1);
+		#else
+			// Path must be absolute
+			ret = LINUX_SYSCALL(__NR_readlinkat, -1, buf, path, sizeof(path) - 1);
```

Comment:
@LubosD If the path is absolute, would it be fine if I hard code the `dirfd` argument to -1 (invalid value)? Most Linux syscalls that use the `dirfd` argument will ignore it if the path is absolute.

Reply:
Nevermind, found out that `AT_FDCWD` exists.